### PR TITLE
Fix test update branch gray rules with update once

### PR DESCRIPTION
**Pull request by ZhewenFu**
## What's the purpose of this PR

Fix test `testUpdateBranchGrayRulesWithUpdateOnce`

## Which issue(s) this PR fixes:
Fixes the issue that causes flaky tests. The current test may lead to inconsistent ordering of stringified JSON objects and the test may fail sometimes. To fix the test and make it consistent, I select all the elements in the list and see whether the release history contains those elements. In this case, the ordering of the elements doesn't matter and won't cause failure.

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [ ] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [ ] Update the [`CHANGES` log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md).

### CODE REVIEW

File: apollo-biz/src/test/java/com/ctrip/framework/apollo/biz/service/NamespaceBranchServiceTest.java [hunk 1]
```diff
@@ -82,7 +82,15 @@ public void testUpdateBranchGrayRulesWithUpdateOnce() {
     Assert.assertEquals(ReleaseOperation.APPLY_GRAY_RULES, releaseHistory.getOperation());
     Assert.assertEquals(0, releaseHistory.getReleaseId());
     Assert.assertEquals(0, releaseHistory.getPreviousReleaseId());
-    Assert.assertTrue(releaseHistory.getOperationContext().contains(rule.getRules()));
+    Assert.assertTrue(containRule(releaseHistory.getOperationContext(),rule.getRules()));
+  }
+
+  public boolean containRule(String context, String rule){
```
Comment (mghio) [inline apollo-biz/src/test/java/com/ctrip/framework/apollo/biz/service/NamespaceBranchServiceTest.java:88]: How about use `private`?

File: apollo-biz/src/test/java/com/ctrip/framework/apollo/biz/service/NamespaceBranchServiceTest.java [hunk 2]
```diff
@@ -82,7 +82,15 @@ public void testUpdateBranchGrayRulesWithUpdateOnce() {
...
+    boolean contains = true;
+    String[] rules = rule.substring(2,rule.length()-2).split(",");
```
Comment (mghio) [inline apollo-biz/src/test/java/com/ctrip/framework/apollo/biz/service/NamespaceBranchServiceTest.java:90]: I think use [GrayReleaseRuleItemTransformer#batchTransformFromJSON](https://github.com/apolloconfig/apollo/blob/301d66dcc28b2d698eefed3ecc9aa3273afb9ddf/apollo-common/src/main/java/com/ctrip/framework/apollo/common/utils/GrayReleaseRuleItemTransformer.java#L35) parse  rules might be better. What do you think?

**Review by mghio — CHANGES_REQUESTED** [review]
BTW, please use the [apollo code style](https://github.com/apolloconfig/apollo/tree/master/apollo-buildtools/style) for format the codes.

**Comment by ZhewenFu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ZhewenFu** [comment]
>

**Pull request closed without merging**